include/d3dx12Residency.h

```diff
@@ -635,11 +713,18 @@ namespace D3DX12Residency
 						break;
 					}
 
-					RESIDENCY_CHECK(pObject->ResidencyStatus == ManagedObject::RESIDENCY_STATUS::RESIDENT);
-					EvictionList[NumObjectsToEvict++] = pObject->pUnderlying;
-					Evict(pObject);
+					if (pObject->IsPinned())
+					{
+						pResourceEntry = pResourceEntry->Flink;
+					}
+					else
+					{
+						RESIDENCY_CHECK(pObject->ResidencyStatus == ManagedObject::RESIDENCY_STATUS::RESIDENT);

```

Comment:
The previous check for this (line 684) is outside of the pinned check, while this is inside. I think this should probably be outside too.